eventapi.py

```diff
@@ -0,0 +1,60 @@
+import json
+from datetime import datetime
+from flask import request
+from flask_restful import Resource, abort
+
+from event import Event, DAYS
+
+def read_args(request):
+    return json.loads(request.data.decode('utf-8'))
+
+def read_time(time_str):
+    return datetime.strptime(time_str, '%H:%M:%S').time()
+
+def validate_days(days):
+    if not set(days).issubset(DAYS):
+        raise ValueError('Invalid day string(s) in list')
+
+class EventAPI(Resource):
+    # Returns event details in JSON format
+    def get(self, name):
+        e_id, event = Event.get(name)
+        return event.serialize()
+
+    # Creates a new event based on args and name
+    def post(self, name):
+        event_dict = read_args(request)
+        try:
+            event_dict['name'] = name
+            event_dict['from_time'] = read_time(event_dict['from_time'])
+            event_dict['to_time'] = read_time(event_dict['to_time'])
+            if event_dict['from_time'] > event_dict['to_time']:
```

Comment:
you are raising an exception within a try/except to catch an exception raised within the try/except ..🤔.. lets separate the validation logic of the times into that `validations.py` file I mentioned above .. it will not only validate that the time parameter is correct but also one is greater than the other